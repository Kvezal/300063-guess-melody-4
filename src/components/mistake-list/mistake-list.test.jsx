import React from "react";
import render from "react-test-renderer";
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import MistakeList from "./mistake-list";


Enzyme.configure({
  adapter: new Adapter(),
});

describe(`MistakeListComponent`, () => {
  test(`should render component`, () => {
    const mistakeListSnapshot = render.create(<MistakeList mistakes={3} />).toJSON();
    expect(mistakeListSnapshot).toMatchSnapshot();
  });

  test(`should have game__mistakes container`, () => {
    const tree = shallow(<MistakeList mistakes={0}/>);
    const container = tree.find(`div.game__mistakes`);
    expect(container).toHaveLength(1);
  });

  test.each([1, 2, 3])(`should have %p mistakes`, (mistakes) => {
    const tree = shallow(<MistakeList mistakes={mistakes}/>);
    const mistakeList = tree.find(`div.wrong`);
    expect(mistakeList).toHaveLength(mistakes);
  });
});
