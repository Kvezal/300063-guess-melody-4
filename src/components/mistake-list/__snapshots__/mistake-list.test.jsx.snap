// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MistakeListComponent should render component 1`] = `
<div
  className="game__mistakes"
>
  <div
    className="wrong"
  />
  <div
    className="wrong"
  />
  <div
    className="wrong"
  />
</div>
`;
